crates/common/tedge_users/src/unix.rs

```diff
@@ -33,23 +33,25 @@ struct InnerUserManager {
     guard: Option<users::switch::SwitchUserGuard>,
 }
 
-impl UserManager {
+impl Default for UserManager {
```

Comment:
I understand that clippy warns you when you simply make `new` and not take any inputs, but given the constraints imposed on the `new` method I do not think that this should be accepted.

Instead I propose that `new` returns a `Option<UserManager>` which is internally handled by a [`Once`](https://doc.rust-lang.org/std/sync/struct.Once.html) to verify it is truly only created once.

Reply:
It would be really nice to enforce that `new`  is called only once.

As a side remark, this is this kind of questions that gives me a mix feeling when ask to make clippy warnings hard errors.